code/modules/power/supermatter/supermatter.dm

```diff
@@ -639,16 +673,26 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 		if(power_changes)
 			power = max((removed.temperature * temp_factor / T0C) * gasmix_power_ratio + power, 0)
 
+		//Passively gain power when antinoblium_multiplier is above 1.
+		power += (antinoblium_multiplier - 1) * ANTINOBLIUM_POWER_GENERATION
+
 		if(prob(50))
 			//(1 + (tritRad + pluoxDampen * bzDampen * o2Rad * plasmaRad / (10 - bzrads))) * freonbonus
 			radiation_pulse(src, power * max(0, (1 + (power_transmission_bonus/(10-(gas_comp[/datum/gas/bz] * BZ_RADIOACTIVITY_MODIFIER)))) * freonbonus))// RadModBZ(500%)
 
 		if(prob(gas_comp[/datum/gas/zauker]))
 			playsound(src.loc, 'sound/weapons/emitter2.ogg', 100, TRUE, extrarange = 10)
-			supermatter_zap(src, 6, clamp(power*2, 4000, 20000), ZAP_MOB_STUN, zap_cutoff = src.zap_cutoff, power_level = power, zap_icon = src.zap_icon)
-
-		if(gas_comp[/datum/gas/bz] >= 0.4 && prob(30 * gas_comp[/datum/gas/bz]))
-			src.fire_nuclear_particle()        // Start to emit radballs at a maximum of 30% chance per tick
+			supermatter_zap(src, 256, clamp(power*2, 4000, 20000), ZAP_MOB_STUN, zap_cutoff = src.zap_cutoff, power_level = power, zap_icon = src.zap_icon)
```

Comment:
Absolutely not. a range of 256 means worse case iterating over ALL OF THE ITEMS ON THE Z LEVEL. That will lag to absolute hell.

It's not always that expensive because the proc is based on oview(), but that's still not a risk I'm up for taking

Reply:
What would be a more appropriate value? I want it high to make zauker more unique, but not laggy.

Reply:
Alright I set it to 36 for now. In a future PR I might add a limit to number of objects to zap so I can raise this with less lag.

Reply:
That's still really high. Cause `oview()` is worst case senario `orange()`, or a square twice the width and height of the input.
So 36 would be 82x82, so if there's 2 items per tile, `13448` items to iterate over, or pull in the first place.

Limiting the objects to zap wouldn't really help this, because the cost is basically asking byond to give you all the items at once, and from what I understand it doesn't like, give you one, wait for you to ask for another, give you another, etc. it gives them all at once.

Reply:
What's the effect you want to have here? making zauker's zap stand out more?

Reply:
Yeah.